objects/src/transaction/proven_tx.rs

```diff
@@ -76,6 +76,18 @@ impl ProvenTransaction {
         self.block_ref
     }
 
+    /// Returns an iterator over the unauthenticated input notes in this transaction.
+    pub fn get_unauthenticated_notes(&self) -> impl Iterator<Item = NoteId> + '_ {
+        self.input_notes
+            .iter()
+            .filter_map(|note| note.header().map(|header| header.id()))
+    }
```

Comment:
Would it make sense to return a full `NoteHeader` from here? For example:
```rust
pub fn get_unauthenticated_notes(&self) -> impl Iterator<Item = &NoteHeader>
```
If so, I'd change the description to: "Returns an iterator of the headers of unauthenticated input notes in this transaction."